SunEditor.js

```diff
@@ -112,8 +112,7 @@ class SunEditor extends Component {
   componentDidUpdate(prevProps) {
     // Props compared
     if (prevProps.setContents !== this.props.setContents) {
-      this.editor.setContents(this.props.setContents);
-      this.editor.core.context.element.wysiwyg.blur();
+      !this.editor.core.hasFocus && this.editor.setContents(this.props.setContents);
```

Comment:
- I tried `this.editor.core.focusEdge()`, but people may edit from any position.  ❌
- Remove all this `if statement` also works fine, I tested. But people may change contents somewhere outside the editor. ❌

So, I think on editing state, there's no reason  to setContents somewhere else。And without calling `setContents`, the content inside the editor and `onChange` value are still match because editor will set content by itself when typing.

Reply:
When editing, the `hasFocus` should be `true`.

Reply:
Hi i got your idea but the reason why we set the contents on prop change is that sometimes the contents are been set from somewhere else. And if we do do not handle it like this it will never get set. For example imagine that the editor has not focus and setContents is been set by the state and the person set the state after getting response from the server. This will never work.

Reply:
What i did is that when the setContents change the editor should always blur because it will be receiving input from somewhere else while appending the other contents to right

Reply:
Yeah, but it encounter this problem https://github.com/mkhstar/suneditor-react/issues/33#issuecomment-612866728.

Probably  you misunderstood me,  `!this.editor.core.hasFocus` means has no focus,  and setContents would be set.

Reply:
i got it i just missed the !